exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -0,0 +1,248 @@
+	n := ld.ResourceLogs().Len()
+	if n == 0 {
+		return []*cloudwatchlogs.InputLogEvent{}, 0, nil
+	}
+
+	var dropped int
+	out := make([]*cloudwatchlogs.InputLogEvent, 0) // TODO(jbd): set a better capacity
+
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		ills := rl.InstrumentationLibraryLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ils := ills.At(j)
+			logs := ils.Logs()
+			for k := 0; k < logs.Len(); k++ {
+				log := logs.At(k)
+				event, err := logToCWLog(rl.Resource(), log)
+				if err != nil {
+					logger.Debug("Failed to convert to CloudWatch Log", zap.Error(err))
+					dropped++
+				} else {
+					out = append(out, event)
+				}
+			}
+		}
+	}
+	return out, dropped, nil
+}
+
+type cwLogBody struct {
+	Name                   string                 `json:"name,omitempty"`
+	Body                   interface{}            `json:"body,omitempty"`
+	SeverityNumber         int32                  `json:"severity_number,omitempty"`
+	SeverityText           string                 `json:"severity_text,omitempty"`
+	DroppedAttributesCount uint32                 `json:"dropped_attributes_count,omitempty"`
+	Flags                  uint32                 `json:"flags,omitempty"`
+	TraceID                string                 `json:"trace_id,omitempty"`
+	SpanID                 string                 `json:"span_id,omitempty"`
+	Attributes             map[string]interface{} `json:"attributes,omitempty"`
+	Resource               map[string]interface{} `json:"resource,omitempty"`
+}
+
+func logToCWLog(resource pdata.Resource, log pdata.LogRecord) (*cloudwatchlogs.InputLogEvent, error) {
+	// TODO(jbd): Benchmark and improve the allocations.
+	// Evaluate go.elastic.co/fastjson as a replacement for encoding/json.
+	body := &cwLogBody{
+		Name:                   log.Name(),
+		Body:                   attrValue(log.Body()),
+		SeverityNumber:         int32(log.SeverityNumber()),
+		SeverityText:           log.SeverityText(),
+		DroppedAttributesCount: uint32(log.DroppedAttributesCount()),
+		Flags:                  log.Flags(),
+	}
+	if traceID := log.TraceID(); traceID.IsValid() {
+		body.TraceID = traceID.HexString()
+	}
+	if spanID := log.SpanID(); spanID.IsValid() {
+		body.TraceID = spanID.HexString()
+	}
+
+	if log.Attributes().Len() > 0 {
+		attrs := make(map[string]interface{}, log.Attributes().Len())
+		log.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+			attrs[k] = attrValue(v)
+		})
+		body.Attributes = attrs
+	}
+
+	// Add resource attributes.
+	if resource.Attributes().Len() > 0 {
+		attrs := make(map[string]interface{}, log.Attributes().Len())
+		resource.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+			attrs[k] = attrValue(v)
+		})
+		body.Resource = attrs
+	}
+
+	bodyJSON, err := json.Marshal(body)
+	if err != nil {
+		return nil, err
+	}
+	return &cloudwatchlogs.InputLogEvent{
+		Timestamp: aws.Int64(int64(log.Timestamp()) / (1000 * 1000)), // milliseconds
```

Comment:
Is it possible to use `time.Millisecond`?

Reply:
Yo mean, int64(log.Timestamp()) / int64(time.Millisecond)? 

I don't think it's very readable.

Reply:
Well it's self-documenting and compile-time safe. We have a comment for `// milliseconds` anyways so the current code seems to acknowledge it's a bit tricky.